src/pages/comparison/load-manager.ts

```diff
@@ -0,0 +1,48 @@
+export class LoadManager extends EventTarget {
+  private leftResolved = true;
+  private rightResolved = true;
+  private interval;
+  public haveBeenCompared = false;
+  public loadingTime = 0;
+
+  constructor() {
+    super();
+  }
+
+  startLoading() {
+    this.getTime();
+    this.leftResolved = false;
+    this.rightResolved = false;
+    this.haveBeenCompared = true;
+  }
+
+  resolveLeftSide() {
+    this.leftResolved = true;
+    return this.isLoaded();
+  }
+
+  resolveRightSide() {
+    this.rightResolved = true;
+    return this.isLoaded();
+  }
+
+  stoptTimer() {
+    clearInterval(this.interval);
+  }
+
+  private getTime() {
+    const startTime = Date.now();
+    this.interval = setInterval(() => {
```

Comment:
setInterval is bad to measure elapsed time correctly. Just keep start time and calculate diff after loading